src/charm.py

```diff
@@ -18,6 +22,63 @@ class SpringBootCharm(CharmBase):
     def __init__(self, *args):
         """Initialize the instance."""
         super().__init__(*args)
+        self.framework.observe(self.on.config_changed, self.reconciliation)
+
+    def _spring_boot_container(self) -> ops.model.Container:
+        """Retrieve the container for the Spring Boot application.
+
+        Returns:
+            An instance of :class:`ops.charm.Container` represents the Spring Boot container.
+        """
+        container = self.unit.get_container("spring-boot-app")
+        return container
+
+    def _generate_spring_boot_layer(self) -> dict:
+        """Generate Spring Boot service layer for pebble.
+
+        Returns:
+            Spring Boot service layer configuration, in the form of a dict
+        """
+        container = self._spring_boot_container()
+        if not container.isdir("/app"):
+            raise ReconciliationError(new_status=BlockedStatus("dir /app does not exist"))
+        files_in_app = container.list_files("/app")
+        jar_files = [file.name for file in files_in_app if file.name.endswith(".jar")]
+        if not jar_files:
+            raise ReconciliationError(new_status=BlockedStatus("no jar file exists in /app"))
```

Comment:
Same as above: "No jar file found in /app"

Reply:
Updated, thanks!